### ISSUE TITLE
Renaming of channel breaks links/references

### ISSUE THREAD

**Issue by mld18** [RocketChat/Rocket.Chat#7108]
Rocket.Chat Version: 0.55.1
Running Instances: 1
DB Replicaset OpLog: Disabled
Node Version: v4.2.6

**When channels are renamed the links/references that point to the channel stop working. Moreover, the link labels keep the previous channel name.**

E.g., suppose there is a channel #testchannel. In another channel a user wrote "Hey, see discussion in #testchannel". Now, we rename the channel to #a-testchannel. The message text remains as it used to be before the renaming action. Clicking on the #testchannel link in the message now yields an error message: 
![image](https://cloud.githubusercontent.com/assets/23357345/26523811/a686e51a-431f-11e7-9947-617fc9d7e456.png)

**My expectations:**
1. The link in the message is still working (in Mongo DB the message even contains the technical key to the channel!)
2. The message texts are always changed accordingly (=> "Hey, see discussion in #a-testchannel"). Perhaps, the changing user may choose whether the labels of channel references are supposed to be updated or not.

To me, this feels like a bug as it makes renaming a no-go.
This feature is important to us. We have a vast number of channels and seek for a way to flag certain channels. Way to go seems to be suffixes in the channel names. (I'll file a separate feature request for this.)

**Comment by localguru** [comment]
Is the idea behind suffixes organizing channels in categories? See #2049.

**Comment by mld18** [comment]
@localguru: That's right. However, I've opened #7109 for this. I'll link the two issues. Thanks for the hint!

From my perspective this issue remains a bug (renaming doesn't work as expected).

**Comment by mld18** [comment]
A quarter of a year and a couple of Rocket releases have gone by and I wanted to check if there's any news on this bug here?
It's still a real impediment for us. I'm a little surprised that I see lots of new features in the release notes but it seems to me that this core functionality issue has remained untouched for quite a long time.

**Comment by Micha0203** [comment]
And almost one year later still no reply to this?
I just checked if this Bug is solved but we still have the same issue. 

Any Plans on fixing this anytime??

**Comment by Kailash0311** [comment]
I have fixed the issue but I am not able to change the unit tests accordingly.
Help appreciated. :)
